src/main/java/net/runelite/rsb/wrappers/RSObject.java

```diff
@@ -191,12 +194,25 @@ public Type getType() {
 		return type;
 	}
 
+
+	/**
+	 * @param action the object action
+	 * @return <code>true</code> if the action is prevent, <code>false</code> if the object does not contain the
+	 * desired action
+	 */
+	public boolean hasAction(@NonNull final String action) {
```

Comment:
Note I have had some problems with using the definition like this. This should definitely exist and I think it is good how it is though. We shouldn't check this before trying to doAction as the hover menu is much more accurate.

Reply:
I used this to find an open door. Since a closed door is a different object this works perfectly.

It's currently not being used in the OSRSBot code and is simply available to scripters.

In which case does this not work?

Reply:
Some objects don't have actions properly defined in their definitions. The easiest example is any of these https://static.runelite.net/api/runelite-api/net/runelite/api/NullObjectID.html like the levers in draynor manor basement. There are a bunch more though where the definition of actions isn't the same as the object in certain game states.